src/region.js

```diff
@@ -33,155 +33,108 @@ var Region = MarionetteObject.extend({
     MarionetteObject.call(this, options);
   },
 
-  // Displays a backbone view instance inside of the region.
-  // Handles calling the `render` method for you. Reads content
-  // directly from the `el` attribute. Also calls an optional
-  // `onShow` and `onDestroy` method on your view, just after showing
-  // or just before destroying the view, respectively.
-  // The `preventDestroy` option can be used to prevent a view from
-  // the old view being destroyed on show.
-  // The `forceShow` option can be used to force a view to be
-  // re-rendered if it's already shown in the region.
+  // Displays a backbone view instance inside of the region. Handles calling the `render`
+  // method for you. Reads content directly from the `el` attribute. The `preventDestroy`
+  // option can be used to prevent a view from the old view being destroyed on show.
+  // The `forceShow` option can be used to force a view to be re-rendered if it's already
+  // shown in the region.
   show: function(view, options) {
     if (!this._ensureElement()) {
       return;
     }
+    this._ensureView(view);
 
-    this._ensureViewIsIntact(view);
-    MonitorDOMRefresh(view);
-
-    var showOptions     = options || {};
-    var isDifferentView = view !== this.currentView;
-    var forceShow       = !!showOptions.forceShow;
-    var replaceElement  = !!showOptions.replaceElement;
-
-    // We are only changing the view if there is a current view to change to begin with
-    var changingView = this.currentView;
-    var isChangingView = !!changingView;
+    MonitorViewEvents(view);
 
-    // Only destroy the current view if we don't want to `preventDestroy` and if
-    // the view given in the first argument is different than `currentView`
-    var _shouldDestroyView = this.shouldDestroyView(view, options);
-
-    // Only show the view given in the first argument if it is different than
-    // the current view or if we want to re-show the view. Note that if
-    // `_shouldDestroyView` is true, then `_shouldShowView` is also necessarily true.
-    var _shouldShowView = isDifferentView || forceShow;
-
-    // only replace the region's element with the view's element if explicitly set
-    var _shouldReplaceElement = replaceElement;
+    const { forceShow, replaceElement, preventDestroy, triggerAttach } = options || {};
+    const changingView = this.currentView;
+    const isChangingView = !!changingView;
+    const isDifferentView = view !== this.currentView;
+    const shouldForceShow = !!forceShow;
+    const shouldDestroyView = !preventDestroy;
+    const shouldEmptyRegion = isDifferentView && shouldDestroyView;
+    const shouldShowView = isDifferentView || shouldForceShow;
 
     if (isChangingView) {
```

Comment:
`swap` events have been removed after reasoning that exact conditions for a `swap` can be reproduced with show/empty events and condition on `this.currentView`.